hedera-node/docs/design/services/token-service/airdrops/airdrop-transaction.md

```diff
@@ -0,0 +1,208 @@
+
+Note that for all these fees, a given sender, which is the actual token owner, could be covered by a different payer, which is willing to pay. However, in case the receiver has not claimed an airdrop for a more than a full auto-renew cycle, the sender (token owner) would be required to continue paying future rents and they cannot be covered by a separate payer.
+
+An update into the `feeSchedule` file would be needed to specify the addition of the new fee values for the airdrop transaction type itself.
+
+### Keys
+
+Since a given sender/token owner could be the payer of the airdrop transaction or they can rely on a separate payer account we have a couple of use cases related to the key verification logic.
+
+If the sender of the airdrop is also the payer of this transaction, we would need only their key to be present in the transaction context.
+
+If the airdrop transaction fees from the list above are covered by a separate payer, then both the keys of the airdrop sender and the payer should be present in the context. That is needed, so that the sender authorizes giving an airdrop from their own balance.
+
+### Service updates
+
+- Update `TokenServiceDefinition` class to include the new RPC method definition for sending airdrops
+- Implement new `TokenAirdropHandler` class which should be invoked when the gRPC server handles `TokenAirdropTransaction` transactions. The class should be responsible for:
+    - Verify that the pending airdrops list contains between 1 and 10 entries, inclusive
+    - Verify that the pending airdrops list does not have any duplicate entries
+    - Pre-handle:
+        - The transaction must be signed by the sender for each entry in the `token_transfers` list
+        - In the case where the airdrop fees are covered by a separate payer, the transaction must include also the payer’s signature
+    - Handle:
+        - Any additional validation depending on config or state i.e. semantics checks
+        - Check that the airdrop sender account is a valid account. That is:
+          - an existing account, which was not deleted or expired
+          - a non-existing account, pointed by an alias of `evm_address` or public `ECDSA key`
+        - Check that the token being airdropped, does not contain a custom fee, which needs to be covered by the recipient. This includes `fractionalFees` with `net_of_transfers=false` and `royaltyFees`, including `fallbackRoyaltyFees`.
+        - Such transactions should be reverted and rejected.
+        - The business logic for sending pending airdrops
+            - Check the association status and the existence of the recipient:
+
+              - It is an existing one and has `max_auto_associations` set to -1 and is not associated to the token →
+                the token is auto associated and is directly transferred to the recipient and a `TokenTransferList` is added to the `TransactionRecord`, as well as a new `TokenAssociation` entry in the `automatic_token_associations`
```

Comment:
It looks like you're trying to create a case/answer structure with the arrow at end of line, if that's the case, you probably need to add an actual line break ("view" this file to see how github removes the line breaks otherwise).
```suggestion
              - It is an existing one and has `max_auto_associations` set to -1 and is not associated to the token →<br/>
                The token is auto associated and is directly transferred to the recipient and a `TokenTransferList` is added to the `TransactionRecord`, as well as a new `TokenAssociation` entry in the `automatic_token_associations`
```

Reply:
Noted